### PR TITLE
Feature : Delete Specialisation

### LINKED ISSUE #145: As a user, I can remove specialisation for a doctor
So that I can update the specified doctor's specialisation.

### PR DESCRIPTION
**Pull request by kimshitong**
Closes #145

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteSpecialisationCommand.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    public static final String MESSAGE_DELETE_SPECIALISATION_SUCCESS = "Specialisation deleted: %1$s";
+    public static final String MESSAGE_DELETE_SPECIALISATION_FAILURE = "Specialisation does not exist: %1$s";
+
+    private final Index index;
+    private final Tag specialisation;
+
+    /**
+     * @param index of the doctor in the filtered doctor list to delete the specialisation
+     * @param specialisation of the doctor to be updated to
```
Comment (yanghengtang) [inline src/main/java/seedu/address/logic/commands/DeleteSpecialisationCommand.java:38]: `to be updated to` change to `to be deleted`

File: src/test/java/seedu/address/logic/commands/DeleteSpecialisationCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        // ensures that outOfBoundIndex is still in bounds of database list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getDatabase().getDoctorList().size());
+
+        DeleteSpecialisationCommand command =
+                new DeleteSpecialisationCommand(outOfBoundIndex, DERMATOLOGY_TAG);
+        assertCommandFailure(command, model, Messages.MESSAGE_INVALID_DOCTOR_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Edit patient where condition is not already present
```
Comment (yanghengtang) [inline src/test/java/seedu/address/logic/commands/DeleteSpecialisationCommandTest.java:122]: Update javadocs

File: src/test/java/seedu/address/logic/commands/DeleteSpecialisationCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import seedu.address.model.Database;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.doctor.Doctor;
+import seedu.address.model.tag.Tag;
+import seedu.address.testutil.DoctorBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for AddSpecialisationCommand.
```
Comment (mounilsankar) [inline src/test/java/seedu/address/logic/commands/DeleteSpecialisationCommandTest.java:32]: Should be DeleteSpecialisationCommand

File: src/main/java/seedu/address/logic/commands/DeleteSpecialisationCommand.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+ */
+public class DeleteSpecialisationCommand extends Command {
+    public static final String COMMAND_WORD = "delete_spec";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes a specialisation of the doctor "
+            + "by the index number used in the last doctor listing. "
+            + "Parameters: INDEX (must be a positive integer) "
+            + "t\\ [SPECIALISATION]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "t\\ Orthopaedic ";
```
Comment (mounilsankar) [inline src/main/java/seedu/address/logic/commands/DeleteSpecialisationCommand.java:28]: Use PREFIX_TAG

File: src/main/java/seedu/address/logic/commands/DeleteSpecialisationCommand.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.doctor.Doctor;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Deletes a specialisation of an existing doctor in the database.
+ */
+public class DeleteSpecialisationCommand extends Command {
+    public static final String COMMAND_WORD = "delete_spec";
```
Comment (yanghengtang) [inline src/main/java/seedu/address/logic/commands/DeleteSpecialisationCommand.java:22]: we using `delete_tag_d` instead right?

Comment (kimshitong) [reply src/main/java/seedu/address/logic/commands/DeleteSpecialisationCommand.java]: Yes, standardised..

**Review by mounilsankar — APPROVED** [review]

**Pull request merged**